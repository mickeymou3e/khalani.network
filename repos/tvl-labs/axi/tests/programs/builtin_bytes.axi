!eq_chain(3_u8, 3u8);
!eq_chain(0x3344aabb, 0x3344aabb);
!prove_by_eval(bytes_len(BytesCons(2u8, 0x334455)) add 2 = Succ(4 add 1));

!claim(forall b: u8. forall bs: Bytes. bytes_len(BytesCons(b, bs)) = Succ(bytes_len(bs)));
