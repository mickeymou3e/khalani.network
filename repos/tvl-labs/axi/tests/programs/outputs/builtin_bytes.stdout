Result: Proved(=[3_u8, 3_u8])
Result: Proved(=[0x3344aabb, 0x3344aabb])
Result: Proved(=[add[bytes_len[BytesCons[2_u8, 0x334455]):Bytes]):Nat, 2]):Nat, Succ[add[4, 1]):Nat]):Nat])
Result: Proved(forall u8. forall Bytes. =[bytes_len[BytesCons[@1:u8, @0:Bytes]):Bytes]):Nat, Succ[bytes_len[@0:Bytes]):Nat]):Nat])
